src/IRtext.cpp

```diff
@@ -280,134 +280,258 @@ IRTEXT_CONST_STRING(kDg11j13aStr, D_STR_DG11J13A);  ///< "DG11J13A"
-    D_STR_AIRWELL "\x0"
-    D_STR_DELONGHI_AC "\x0"
-    D_STR_DOSHISHA "\x0"
-    D_STR_MULTIBRACKETS "\x0"
-    D_STR_CARRIER_AC40 "\x0"
-    D_STR_CARRIER_AC64 "\x0"
-    D_STR_HITACHI_AC344 "\x0"
-    D_STR_CORONA_AC "\x0"
-    D_STR_MIDEA24 "\x0"
-    D_STR_ZEPEAL "\x0"
-    D_STR_SANYO_AC "\x0"
-    D_STR_VOLTAS "\x0"
-    D_STR_METZ "\x0"
-    D_STR_TRANSCOLD "\x0"
-    D_STR_TECHNIBEL_AC "\x0"
-    D_STR_MIRAGE "\x0"
-    D_STR_ELITESCREENS "\x0"
-    D_STR_PANASONIC_AC32 "\x0"
-    D_STR_MILESTAG2 "\x0"
-    D_STR_ECOCLIM "\x0"
-    D_STR_XMP "\x0"
-    D_STR_TRUMA "\x0"
-    D_STR_HAIER_AC176 "\x0"
-    D_STR_TEKNOPOINT "\x0"
-    D_STR_KELON "\x0"
-    D_STR_TROTEC_3550 "\x0"
-    D_STR_SANYO_AC88 "\x0"
-    D_STR_BOSE "\x0"
-    D_STR_ARRIS "\x0"
-    D_STR_RHOSS "\x0"
-    D_STR_AIRTON "\x0"
-    D_STR_COOLIX48 "\x0"
-    D_STR_HITACHI_AC264 "\x0"
-    D_STR_KELON168 "\x0"
-    D_STR_HITACHI_AC296 "\x0"
-    D_STR_DAIKIN200 "\x0"
-    D_STR_HAIER_AC160 "\x0"
-    D_STR_CARRIER_AC128 "\x0"
-    D_STR_TOTO "\x0"
-    D_STR_CLIMABUTLER "\x0"
-    D_STR_TCL96AC "\x0"
-    D_STR_BOSCH144 "\x0"
-    D_STR_SANYO_AC152 "\x0"
-    D_STR_DAIKIN312 "\x0"
-    ///< New protocol strings should be added just above this line.
+    COND(DECODE_RC5 || SEND_RC5,
+            D_STR_RC5, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_RC6 || SEND_RC6,
+            D_STR_RC6, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_NEC || SEND_NEC,
+            D_STR_NEC, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_SONY || SEND_SONY,
+            D_STR_SONY, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_PANASONIC || SEND_PANASONIC,
+            D_STR_PANASONIC, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_JVC || SEND_JVC,
+            D_STR_JVC, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_SAMSUNG || SEND_SAMSUNG,
+            D_STR_SAMSUNG, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_WHYNTER || SEND_WHYNTER,
+            D_STR_WHYNTER, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_AIWA_RC_T501 || SEND_AIWA_RC_T501,
+            D_STR_AIWA_RC_T501, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_LG || SEND_LG,
+            D_STR_LG, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_SANYO || SEND_SANYO,
+            D_STR_SANYO, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_MITSUBISHI || SEND_MITSUBISHI,
+            D_STR_MITSUBISHI, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_DISH || SEND_DISH,
+            D_STR_DISH, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_SHARP || SEND_SHARP,
+            D_STR_SHARP, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_COOLIX || SEND_COOLIX,
+            D_STR_COOLIX, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_DAIKIN || SEND_DAIKIN,
+            D_STR_DAIKIN, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_DENON || SEND_DENON,
+            D_STR_DENON, D_STR_UNSUPPORTED) "\x0"
+    COND(DECODE_KELVINATOR || SEND_KELVINATOR,
+            D_STR_KELVINATOR, D_STR_UNSUPPORTED) "\x0"
+    COND(SEND_SHERWOOD,
+            D_STR_SHERWOOD, D_STR_UNSUPPORTED) "x0"  // SEND-ONLY
```

Comment:
Found it, missing a `\` here, should be `"\x0"`, used a keyboard macro to break the lines to the correct length, but that sort-of failed on the first line with a trailing comment...